internal/service/resourcepolicy/resource.go

```diff
@@ -0,0 +1,175 @@
+package resourcepolicy
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+const resourceName = "resource_policy"
+
+var _ resource.ResourceWithConfigure = &resourcePolicyRS{}
+var _ resource.ResourceWithImportState = &resourcePolicyRS{}
+
+var (
+	errorCreate = "error creating resource"
+	errorRead   = "error reading resource"
+	errorUpdate = "error updating resource"
+)
+
+func Resource() resource.Resource {
+	return &resourcePolicyRS{
+		RSCommon: config.RSCommon{
+			ResourceName: resourceName,
+		},
+	}
+}
+
+type resourcePolicyRS struct {
+	config.RSCommon
+}
+
+func (r *resourcePolicyRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = ResourceSchema(ctx)
+}
+
+func (r *resourcePolicyRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan TFResourcePolicyModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	orgID := plan.OrgID.ValueString()
+	policies, diags := NewTFPoliciesModelToSDK(ctx, plan.Policies)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	resourcePolicyAPI := connV2.AtlasResourcePoliciesApi
+	policySDK, _, err := resourcePolicyAPI.CreateAtlasResourcePolicy(ctx, orgID, &admin.ApiAtlasResourcePolicyCreate{
```

Comment:
knit: i suppose it's a matter of preference, I prefer to have the API call in our line, so would create the api request param outside

Reply:
inlined the `api`. Created the param inline too. I think it is readable. And another variable named `req` can be confusing